src/dbus/abrt_problems2_service.c

```diff
@@ -1125,11 +1122,10 @@ static GVariant *entry_object_dbus_get_property(GDBusConnection *connection,
         GVariantBuilder builder;
         g_variant_builder_init(&builder, G_VARIANT_TYPE("as"));
         dd_init_next_file(dd);
-        char *short_name;
+        g_autofree char *short_name = NULL;
         while (dd_get_next_file(dd, &short_name, NULL))
         {
             g_variant_builder_add(&builder, "s", short_name);
-            free(short_name);
         }
```

Comment:
I don't think this is equivalent. In the original code, `short_name` was freed in every iteration, now only the last value is freed once it goes out of scope, so there will be a leak.